crucible-llvm/src/Lang/Crucible/LLVM/Printf.hs

```diff
@@ -170,19 +170,19 @@ formatSignedDec i minwidth prec flags = do
                | Set.member PrintfPosSpace flags -> " "
                | otherwise -> ""
   let digits = N.showInt (abs i) []
-  let precdigits = if prec > 0 then
-                      let n = max 0 (prec - length digits) in
-                      replicate n '0' ++ digits
-                   else
-                      digits
+  let precdigits = case prec of
```

Comment:
The code which is used to compute `precdigits` is repeated four different times in this module, which suggests that it should be factored out into its own function. (There are other similarities among the various `format*` functions, but it's not clear if one can factor them out as cleanly.)